pkg/generator/helper/docuHelper.go

```diff
@@ -107,144 +76,323 @@ func readAndAdjustTemplate(docFile io.ReadCloser) string {
 }
 
 // Replaces the docGenStepName placeholder with the content from the yaml
-func docGenStepName(stepData config.StepData) string {
+func docGenStepName(stepData *config.StepData) string {
 	return stepData.Metadata.Name
 }
 
 // Replaces the docGenDescription placeholder with content from the yaml
-func docGenDescription(stepData config.StepData) string {
-	return "Description\n\n" + stepData.Metadata.LongDescription
+func docGenDescription(stepData *config.StepData) string {
+	description := ""
+
+	description += "Description\n\n" + stepData.Metadata.LongDescription + "\n\n"
+
+	description += "## Usage\n\n"
+	description += "We recommend to define values of [step parameters](#parameters) via [config.yml file](../configuration.md). In this case, calling the step is reduced to one simple line.<br />Calling the step can be done either via the Jenkins library step or on the command line.\n\n"
+	description += "### Jenkins pipelines\n\n```groovy\n"
+	description += fmt.Sprintf("%v script: this\n```\n", stepData.Metadata.Name)
+	description += "### Command line\n\n```\n"
+	description += fmt.Sprintf("piper %v\n```\n\n", stepData.Metadata.Name)
+	description += stepOutputs(stepData)
+	return description
+}
+
+func stepOutputs(stepData *config.StepData) string {
```

Comment:
What do you think about moving the code for each section into a dedicated file to keep the code small and focused?

Reply:
not sure if we need this. We are talking about the doc generator ...
If someone wants to invest in further clean-up I would suggest to do this in a follow-up PR. I just kept the structure in place as it was.